# HG changeset patch
# User Punarbasu Purkayastha <ppurka@gmail.com>
# Date 1381856144 -28800
# Node ID 69c1cf4a08321aa2b9e47a0a912adf0fdbc251b8
# Parent  3c901109788d906f8550daf049525dd4919c843c
allow weight_enumerator to take the variable names separately

diff --git a/sage/coding/linear_code.py b/sage/coding/linear_code.py
--- a/sage/coding/linear_code.py
+++ b/sage/coding/linear_code.py
@@ -2770,14 +2770,19 @@
         V = VectorSpace(F,n+1)
         return V(self.spectrum()).support()
 
-    def weight_enumerator(self, names="xy"):
+    def weight_enumerator(self, names="xy", name2=None):
         """
         Returns the weight enumerator of the code.
 
         INPUT:
 
         - ``names`` - String of length 2, containing two variable names
-          (default: ``"xy"``)
+          (default: ``"xy"``). Alternatively, it can be a variable name or
+          a string, or a tuple of variable names or strings.
+
+        - ``name2`` - string or symbolic variable (default: ``None``).
+          If ``name2`` is provided then it is assumed that ``names``
+          contains only one variable.
 
         OUTPUT:
 
@@ -2790,7 +2795,18 @@
             x^7 + 7*x^4*y^3 + 7*x^3*y^4 + y^7
             sage: C.weight_enumerator(names="st")
             s^7 + 7*s^4*t^3 + 7*s^3*t^4 + t^7
+            sage: (var1, var2) = var('var1, var2')
+            sage: C.weight_enumerator((var1, var2))
+            var1^7 + 7*var1^4*var2^3 + 7*var1^3*var2^4 + var2^7
+            sage: C.weight_enumerator(var1, var2)
+            var1^7 + 7*var1^4*var2^3 + 7*var1^3*var2^4 + var2^7
+
         """
+        if name2 is not None:
+            # We assume that actual variable names or strings are provided
+            # for names if names2 is also provided. That is, names is not
+            # a tuple or a list. Otherwise, PolynomialRing will return error
+            names = (names, name2)
         spec = self.spectrum()
         n = self.length()
         R = PolynomialRing(QQ,2,names)
